vyper/ast/annotation.py

```diff
@@ -38,24 +62,42 @@ def visit_ClassDef(self, node):
 
         return node
 
+    def visit_Constant(self, node):
+        self.generic_visit(node)
+
+        # special case to deal with Constant type in Python >=3.8
+        if node.value is None or isinstance(node.value, bool):
+            node.ast_type = "NameConstant"
+        elif isinstance(node.value, (int, float)):
+            node.ast_type = "Num"
+        elif isinstance(node.value, str):
+            node.ast_type = "Str"
+        elif isinstance(node.value, bytes):
+            node.ast_type = "Bytes"
+        else:
+            raise SyntaxException(f"Invalid syntax (unsupported Python Constant AST node).", node)
+
+        return node
 
-class RewriteUnarySubVisitor(python_ast.NodeTransformer):
     def visit_UnaryOp(self, node):
         self.generic_visit(node)
-        if isinstance(node.op, python_ast.USub) and isinstance(node.operand, python_ast.Num):
+        # NOTE: This is done so that decimal literal now sees the negative sign as part of it
+        if isinstance(node.op, python_ast.USub) and isinstance(
```

Comment:
stylistically I prefer to use variables to shorten the if condition:
```suggestion
is_usub = isinstance(node.op, python_ast.USub)
is_num = isinstance(node.operand, python_ast.Num)
if is_usub and is_num:
  ...
```